tests/unit/pubsub.tcl

```diff
@@ -514,4 +514,79 @@ start_server {tags {"pubsub network"}} {
         assert_equal [r read] {message foo vaz}
     } {} {resp3}
 
+    test "SUBSCRIBE and UNSUBSCRIBE with multiple channels" {
+        # Note: this is testing whether the client exits pubsub mode when subscribed to 0 channels.
+        set rd1 [valkey_deferring_client]
+        
+        assert_equal {1 2 3} [subscribe $rd1 {chan1 chan2 chan3}]
```

Comment:
There is a problem with this test: It doesn't test `valkey-cli`. The client used here is the TCL client which is used internally in the test framework.

The tests for `valkey-cli` are in the file `tests/integration/valkey-cli.tcl`.

Reply:
I deleted the three test cases in the pubsub.tcl file since they didn't test the client and then added one test case to the valkey-cli.tcl file that tests all three commands at once.